### PR TITLE
feat: Free bookings and other stuff

### LINKED ISSUE #190: Handle free bookings
When a booking's total comes to 0, payBooking migration should just set the status of the booking to Paid.

### PR DESCRIPTION
**Pull request by atoff**
Allows for free bookings (by not creating a payment/parsing the payment provider)
Also enforces the need for a booking to have at least one ticket to be paid for

Closes #190

### CODE REVIEW

File: uobtheatre/bookings/schema.py [hunk 1]
```diff
@@ -634,34 +634,43 @@ def resolve_mutation(  # pylint: disable=too-many-arguments
...
-                raise GQLException(
-                    message=f"A device_id is required when using {payment_provider} provider.",
-                    field="device_id",
-                    code="missing_required",
+        # Booking must have at least one ticket
+        if booking.tickets.count() == 0:
+            raise GQLException(message="The booking must have at least one ticket")
+
+        # If the booking isn't free, we care about the payment provider. Otherwise, we don't
+        if booking.total() > 0:
```
Comment (JElgar) [inline uobtheatre/bookings/schema.py:642]: Can we not nest this? So if just the first branch in the `elif`?
```suggestion
        if booking.total() == 0:
            # Free bookings do not require a payment
            pass
```

File: uobtheatre/bookings/test/test_mutations.py [hunk 2]
```diff
@@ -21,6 +22,19 @@
 from uobtheatre.venues.test.factories import SeatFactory, SeatGroupFactory
 
 
+def add_ticket_to_booking(booking):
```
Comment (JElgar) [inline uobtheatre/bookings/test/test_mutations.py:25]: Can we move this into bookings/test/factories

Comment (JElgar) [reply uobtheatre/bookings/test/test_mutations.py:25]: Maybe with some (optional?) arguments?

**Review by JElgar — CHANGES_REQUESTED** [review]

File: uobtheatre/bookings/test/test_mutations.py (hunk 2 above)
Comment (atoff) [reply uobtheatre/bookings/test/test_mutations.py:25]: if we need to extend later with arguments we can, I think for now this should be enough, its mainly just to make it a valid booking

**Review by JElgar — APPROVED** [review]
LGTM

**Pull request merged**